src/main/java/build/buildfarm/instance/shard/ShardInstance.java

```diff
@@ -2634,10 +2641,22 @@ public void deregisterWorker(String workerName) {
     }
   }
 
+  @VisibleForTesting
+  BuildfarmConfigs getBuildFarmConfigs() {
+    return configs;
+  }
+
   private boolean inDenyList(RequestMetadata requestMetadata) throws IOException {
     if (!useDenyList) {
       return false;
     }
     return backplane.isBlacklisted(requestMetadata);
   }
+
+  private Set<String> getActiveWorkerCopy() throws IOException {
```

Comment:
This should probably be switched over in the backplane so that it returns a new HashSet copy each time that can safely be observed - cost should be minimal in creating that new object, and we're doing it by ourselves here. It will make that interface more threadsafe rather than exposing the multiply referenced worker set implementation detail.

Reply:
Thanks for feedback, updated as per request.